### PR TITLE
[wip] Add comments to configuration file

### PR DESCRIPTION
**Pull request by calmh**
This results in this in the config file:

```xml
   ...
   <options>
        <!-- listenAddress: An address (URL format) that we listen on for incoming sync connections. Supports the tcp:// scheme only. -->
        <listenAddress>tcp://0.0.0.0:22000</listenAddress>
        <!-- globalAnnounceServer: An address (URL format) to a global discovery server. Uses udp4:// or udp6:// schemes. -->
        <globalAnnounceServer>udp4://announce.syncthing.net:22027</globalAnnounceServer>
        <globalAnnounceServer>udp6://announce-v6.syncthing.net:22027</globalAnnounceServer>
        <!-- globalAnnounceEnabled: When set to true, global discovery servers are announced to and queried for dynamic addresses. -->
        <globalAnnounceEnabled>true</globalAnnounceEnabled>
   ...
```

This is not pretty, is there a nicer way?

### CODE REVIEW

File: lib/config/config.go [hunk 1]
```diff
@@ -214,9 +214,9 @@ type FolderDeviceConfiguration struct {
 }
 
 type OptionsConfiguration struct {
-	ListenAddress           []string `xml:"listenAddress" json:"listenAddress" default:"tcp://0.0.0.0:22000"`
-	GlobalAnnServers        []string `xml:"globalAnnounceServer" json:"globalAnnounceServers" json:"globalAnnounceServer" default:"udp4://announce.syncthing.net:22027, udp6://announce-v6.syncthing.net:22027"`
-	GlobalAnnEnabled        bool     `xml:"globalAnnounceEnabled" json:"globalAnnounceEnabled" default:"true"`
+	ListenAddress           []string `xml:"listenAddress" json:"listenAddress" default:"tcp://0.0.0.0:22000" comment:"An address (URL format) that we listen on for incoming sync connections. Supports the tcp:// scheme only."`
+	GlobalAnnServers        []string `xml:"globalAnnounceServer" json:"globalAnnounceServers"default:"udp4://announce.syncthing.net:22027, udp6://announce-v6.syncthing.net:22027"comment:"An address (URL format) to a global discovery server. Uses udp4:// or udp6:// schemes."`
```
Comment (AudriusButkevicius) [inline lib/config/config.go]: Missing space before comment

Comment (calmh) [inline lib/config/config.go]: Indeed. And missing comments for all the other fields. :)

**Comment by calmh** [comment]
BTW the "not pretty" comment is about the horrible MarshalXML method, not the config output (which is pretty!).

**Comment by AudriusButkevicius** [comment]
Surprisingly, I was thinking about a similar thing, but also overrriding MarshalJSON to export verbose descriptions to the advanced config view in the web ui.

**Pull request closed without merging**